src/ParametersHandler/tests/ParametersHandlerTest.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file ParametersHandlerTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// std
+#include <any>
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+
+// Catch2
+#include <catch2/catch.hpp>
+
+#include <BipedalLocomotionControllers/ParametersHandler/IParametersHandler.h>
+
+using namespace BipedalLocomotionControllers::ParametersHandler;
+
+// example of parameters handler
+class BasicImplementation : public IParametersHandler<BasicImplementation>
+{
+    std::unordered_map<std::string, std::any> m_map;
+
+public:
+    BasicImplementation(const std::unordered_map<std::string, std::any>& map)
+        : m_map{map}
+    {}
+
+    BasicImplementation() = default;
+
+    template <typename T> void setParameter(const std::string& parameterName, const T& parameter)
+    {
+        m_map.insert({parameterName, std::make_any<T>(parameter)});
```

Comment:
I know it's just a test, but can you use the [``operator[]``](http://www.cplusplus.com/reference/unordered_map/unordered_map/operator[]/) to deal also with the case where the parameter already exists?

Reply:
Yes! Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/13